GP-Engine-team/GP_Engine
# Window interaction with inputs fixed

## Description
This brings several corrective patches to some minor bugs:
- The game viewer window can now be moved around even when the game is being simulated
- When pressing the "play" button, the cursor and the inputs are left configured for the editor. The game simulation captures the inputs only when the user left clicks in the game viewer window
- The game's window now properly shows up and is maximized when GPLauncher is set as default startup project
- While the game is played or paused, the user must press `Ctrl + Shift + S` instead of `Ctrl + S` to prevent any accidental manipulation
- Added a few small but useful methods to `Window`

## How to test
1. Hide the game viewer (set it in a tab next to the scene editor, for instance)
2. Press the "play" button, see that the cursor can still be used to click around in the editor
3. Open and left click in the game viewer, check that the inputs are captured, and that they are released when pressing Esc.
4. Try `Ctrl + S`, check that it does nothing. Try `Ctrl + Shift + S`, and check that it saves the level
4. Close the project, set GPLauncher as startup project, and check that the game shows up and runs in its window
